lib/cpp/src/thrift/transport/TServerSocket.cpp

```diff
@@ -16,7 +16,9 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
+#if defined(_MSC_VER) && (_MSC_VER < 1900)
+#define snprintf _snprintf
```

Comment:
Then I would suggest updating the code that makes THRIFT_SNPRINTF in PlatformSocket.h to be correct for VS2013 and VS2015, and then changing the code that uses snprintf directly to use THRIFT_SNPRINTF.  We should keep this type of platform abstraction in one place, in PlatformSocket.h.